torch/csrc/autograd/functions/convolution.cpp

```diff
@@ -109,48 +110,44 @@ auto ConvForward::output_size(Tensor& input, Tensor& weight) -> std::vector<long
   return output_size;
 }
 
-static auto view4d(const Tensor& tensor) -> std::unique_ptr<Tensor> {
-  if (tensor.nDim() != 3) throw std::runtime_error("expected 3D tensor");
-  auto result = tensor.newTensor();
-  result->unsqueeze(tensor, 2);
-  return result;
+static auto view4d(const at::Tensor& tensor) -> at::Tensor {
+  if (tensor.ndimension() != 3) throw std::runtime_error("expected 3D tensor");
+  return tensor.unsqueeze(2);
 }
 
-static auto view3d(const Tensor& tensor) -> std::unique_ptr<Tensor> {
-  if (tensor.nDim() != 4) throw std::runtime_error("expected 4D tensor");
-  auto result = tensor.newTensor();
-  result->squeeze(tensor, 2);
-  return result;
+static auto view3d(const at::Tensor& tensor) -> at::Tensor {
+  if (tensor.ndimension() != 4) throw std::runtime_error("expected 4D tensor");
+  return tensor.squeeze(2);
 }
 
-static std::unique_ptr<Tensor> subtensor(Tensor* tensor, int dim, int groups, int g) {
-  if (!tensor) {
-    return nullptr;
+static at::Tensor subtensor(at::Tensor& tensor, int dim, int groups, int g) {
+  if (!tensor.defined()) {
+    at::Tensor empty;
+    return empty;
```

Comment:
`at::Tensor()` should work, we can also define `const Tensor kUndefinedTensor = Tensor()` somewhere in ATen to make the intent clearer.